### DISCUSSION TITLE
Homing Z

### DISCUSSION THREAD

**Discussion by moofassah** [Jyers/Marlin#161] [Q&A] [answered]
I have a BL touch previously confirmed working with bugfix branch of the standard Marlin build. 
4.2.7 board
The Z axis will not home.
When I go to Prepare -> Z offset - Z home
The printer homes x/y then moves to bed center, raises z then deploys the probe several times but the Z screw doesn't move down at all to sense the bed.
When done a Message Reading "Stopped" appears at the bottom.

I've just updated to the pre-compiled 427 HS BL touch bin and non HS with no difference.

I must be missing something. 
Any ideas? Do you need more information?

Thank you.

**Comment by Jyers** [comment, marked as answer]
This sounds like a configuration issue of some kind? How do you have your BL Touch connected to the board? As this may result in the board not being able to properly communicate with the probe.

**Reply by moofassah** [reply]
Sorry I should have replied earlier. I figured out the pre-compiled assumes that I have the BLTouch plugged into the 5 pin port and not the Z end stop. Changed the config and works fine.
Thank you for the reply!

**Reply by Jyers** [reply]
Glad you got it working!

**Reply by amherables** [reply]
hello, i have the same problem/same set-up, but have no experience in config and compiling on my own. how can i proceed? not sure what to change in the code to make it look at the z-end stop. I am not using the 5pin port but using an adaptor to connect to z-end stop. thanks.

**Reply by Jyers** [reply]
In configuration.h change "//#define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN" to "#define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN"

**Reply by amherables** [reply]
thank you.

**Comment by moofassah** [comment]
The pre compiled software assumes that you have hooked up you bltouch with all 5 wires on the bl touch port.

This is instead of connecting the black and white wire from the bltouch to the zstop port of the main board.

**Reply by amherables** [reply]
thanks. yes, someone shared the line to update already. now learning to compile. =)
